syntax = "proto3";

option optimize_for = LITE_RUNTIME;

message texture_proto_mssg
{
	enum TYPE
	{
		JPEG = 0;
		PNG = 1;
		BMP = 2;
		GIF = 3;
		TGA = 4;
	};

	enum MAG_FILTER
	{
		MAG_NEAREST = 0;
		MAG_LINEAR = 1;
	};

	enum MIN_FILTER
	{
		MIN_NEAREST = 0;
		MIN_LINEAR = 1;
		MIN_NEAREST_MIPMAP_NEAREST = 2;
		MIN_LINEAR_MIPMAP_NEAREST = 3;
		MIN_NEAREST_MIPMAP_LINEAR = 4;
		MIN_LINEAR_MIPMAP_LINEAR = 5;
	};

	enum WRAP
	{
		CLAMP_TO_EDGE = 0;
		MIRRORED_REPEAT = 1;
		REPEAT = 2;
	};

	enum COMPONENT
	{
		BYTE = 0;
		UNSIGNED_BYTE = 1;
		SHORT = 2;
		UNSIGNED_SHORT = 3;
		INT = 4;
		UNSIGNED_INT = 5;
		FLOAT = 6;
		DOUBLE = 7;
	};

	enum EFORMAT
	{
		EFORMAT_BGR = 0;
		EFORMAT_RGB = 1;
		EFORMAT_RGBA = 2;
	};

	enum NCOMPONENT
	{
		NCOMPONENT_BGR = 0;
		NCOMPONENT_RGB = 1;
		NCOMPONENT_RGBA = 2;
	}


	TYPE			type			= 1;
	MAG_FILTER		magFilter		= 2;
	MIN_FILTER		minFilter		= 3;
	WRAP			wrapS			= 4;
	WRAP			wrapT			= 5;
	COMPONENT		pixelType		= 6;
	EFORMAT			eformat			= 7;
	NCOMPONENT		ncomponent		= 8;
	
	bool			enabled			= 9;
	bool			as_is			= 10;
	
	uint32			width			= 11;
	uint32			height			= 12;
	uint32			component		= 13;
	uint32			bits			= 14;
	uint32			dataSize		= 15;
	uint32			hash			= 16;

	bytes			filename		= 17;
	bytes			pData			= 18;
}
